browser-test/playwright.config.ts

```diff
@@ -25,8 +33,10 @@ export default defineConfig({
     },
   },
   use: {
-    trace: process.env.CI === 'true' ? 'on-first-retry' : 'on',
-    video: process.env.RECORD_VIDEO === 'true' ? 'on-first-retry' : 'off',
+    // retain-on-failure should remove trace and video files when a test succeeds allowing
+    // for minimizing cloud storage needs.
+    trace: process.env.CI === 'true' ? 'retain-on-failure' : 'on',
+    video: process.env.RECORD_VIDEO === 'true' ? 'retain-on-failure' : 'off',
```

Comment:
When the `RECORD_VIDEO` environment variable is true, the test records the video but discards if the test succeed. That way when the artifact upload step occurs it will only have videos of failed tests to save. This is a github cloud storage saving measure. 

The outcome is the same as before. But instead of running the test, failing running again in recording mode it records on the first run. There is some overhead in recording all the time, but given our test run time it should be negligible.